File: solutions/javascript/raindrops/4/raindrops.js
const FACTOR_SOUNDS = [
  [3, 'Pling'],
  [5, 'Plang'],
  [7, 'Plong']
];

export const convert = input => {
  let result = '';

  FACTOR_SOUNDS.forEach((sound) => {
    if (input % sound[0] === 0) {
      result += sound[1];
    }
  });

  return result.length > 0 ? result : input.toString();
};

File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"
#include <string>

namespace raindrops {
    std::string convert (int number) {
        std::string sounds = "";

        if (number % 3 == 0) {
            sounds = sounds + "Pling";
        }

        if (number % 5 == 0) {
            sounds = sounds + "Plang";
        }

        if (number % 7 == 0) {
            sounds = sounds + "Plong";
        }

        if (sounds == "") {
            sounds = std::to_string(number);
        }

        return sounds;
    }
} 
